pkg/operator/ceph/cluster/controller.go

```diff
@@ -254,57 +255,44 @@ func (c *ClusterController) configureExternalCephCluster(namespace, name string,
 	c.updateClusterStatus(namespace, name, cephv1.ClusterStateConnecting, "")
 
 	// loop until we find the secret necessary to connect to the external cluster
-	for {
-		var err error
-		cluster.Info, _, _, err = mon.LoadClusterInfo(c.context, namespace)
-		if err != nil {
-			logger.Warningf("waiting for the connection info of the external cluster. %+v", err)
-			time.Sleep(10 * time.Second)
-			continue
-		} else {
-			break
-		}
-	}
-
-	logger.Infof("found the cluster info to connect to the external cluster. mons=%+v", cluster.Info.Monitors)
+	// then populate clusterInfo
+	cluster.Info = populateExternalClusterInfo(c.context, namespace)
 
-	// Write the rook-ceph-config configmap (used by various daemons to apply config overrides)
-	// If we don't do this, daemons will never start, waiting forever for this configmap to be present
-	err = config.GetStore(cluster.context, namespace, &cluster.ownerRef).CreateOrUpdate(cluster.Info)
+	// Validate versions (local and external)
+	_, _, err = c.detectAndValidateCephVersion(cluster, cluster.Spec.CephVersion.Image)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to detect and validate ceph version. %+v", err)
 	}
 
-	// Let's write connection info (ceph config file and keyring) to the operator for health checks
-	err = mon.WriteConnectionConfig(cluster.context, cluster.Info)
-	if err != nil {
-		return fmt.Errorf("failed to write connection info %+v", err)
+	// Create a configmap 'rook-config-override' for overriding ceph config settings
+	// These settings should only be modified by a user after they are initialized
+	placeholderConfig := map[string]string{
+		k8sutil.ConfigOverrideVal: "",
 	}
-
-	// Get Ceph monitors version on the external cluster
-	cephMonVersion, err := client.GetCephMonVersion(c.context, namespace)
-	if err != nil {
-		return fmt.Errorf("failed to get ceph mon version. %+v", err)
+	cm := &v1.ConfigMap{
```

Comment:
This configmap is also created in `doOrchestration()` in cluster.go. Can we factor it out to a common method?